backend/fms_core/import_tool/importers/experiment_run.py

```diff
@@ -0,0 +1,138 @@
+from pandas import pandas as pd
+from fms_core.models import ExperimentType, PropertyType
+from ._generic import GenericImporter
+from fms_core.import_tool.handlers import ExperimentRunHandler
+from .._utils import (data_row_ids_range,
+                      convert_property_value_to_str,
+                      blank_and_nan_to_none)
+
+
+class ExperimentRunImporter(GenericImporter):
+    base_errors = []
+    global_data = {'experiment_type': None, 'protocols_dict': {}, 'property_types_by_name': {}}
+
+    def __init__(self):
+        self.sheet_names = ['Experiments', 'Samples']
+        super().__init__()
+
+    # has to be declared ?? or can directly inherit from its parent?
+    def import_template(self, file, format, dry_run):
+        result = super().import_template(file, format, dry_run)
+        return result
+
+
+    def import_global_data_from_template(self, workflow, properties):
+        # Preload ExperimentType and Protocols dict
+        try:
+            self.global_data['experiment_type'] = ExperimentType.objects.get(workflow=workflow)
+
+            # Preload Protocols objects for this experiment type in a dictionary for faster access
+            self.global_data['protocols_dict'] = self.global_data['experiment_type'].get_protocols_dict
+        except Exception as e:
+            self.base_errors.append(f"No experiment type with workflow {workflow} could be found.")
+
+        # Preload PropertyType objects for this experiment type in a dictionary for faster access
+        for property_column in properties:
+            try:
+                self.global_data['property_types_by_name'][property_column] = PropertyType.objects.get(
+                    name=property_column)
+            except Exception as e:
+                self.base_errors.append(f"Property Type {property_column} could not be found")
+
+
+    def import_template_inner(self):
+        """
+            SAMPLES SHEET
+        """
+        samples_sheet = self.sheets['Samples']
```

Comment:
I would move most variables to local variables and pass them as method params ... The reason in import export we were relying on the class attribute was that we could not modify class signature without potential consequences ... We do not have this restriction here. Local variables are always more desirable than global ones. All template will have sheets you can pass sheets to import_template_inner...

Reply:
Sheets and sheet_names have to be class variables in order for them to be handled automatically by the GenericImporter (which automatically slice the sheets, read them and store them), rather than repeating this section of code for each Importer.

Reply:
I see a reason for sheet_names but you can just pass sheets to import_template_inner. I would replace individual Sheet_name with a const to make sure all reference to them are correct.